examples/go/main.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 Canonical Ltd.
+// See LICENSE file for licensing details.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"go-app/internal/service"
+	"io"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"encoding/json"
+	"net/http"
+
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type mainHandler struct {
+	counter prometheus.Counter
+	service service.Service
+}
+
+func (h mainHandler) serveHelloWorld(w http.ResponseWriter, r *http.Request) {
+	h.counter.Inc()
+	log.Printf("Counter %#v\n", h.counter)
+	fmt.Fprintf(w, "Hello, World!")
+}
+
+func (h mainHandler) serveEnvs(w http.ResponseWriter, r *http.Request) {
+	h.counter.Inc()
+
+	type EnvVar struct {
+		Name, Value string
+	}
+
+	envVars := []EnvVar{}
+
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		log.Printf("pair 1: %s pair2 %s\n", pair[0], pair[1])
+		envVars = append(envVars, EnvVar{
+			Name:  pair[0],
+			Value: pair[1]})
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(envVars)
+}
```

Comment:
I know this is just an example, but if someone were to copy this and forget to remove it, it would be a pretty bad security issue. Maybe add a note or alert saying that?

Reply:
I have removed that code. It was not used. Instead I have added an endpoint that returns a "hardcoded" env variable, so the risk is removed. This endpoint is now used in an integration test to check used defined variables.